cmake/scripts/buildinfo.cmake

```
@@ -19,19 +19,32 @@ if (NOT ETH_BUILD_PLATFORM)
 	set(ETH_BUILD_PLATFORM "unknown")
 endif()
 
-# Logic here: If prerelease.txt exists but is empty, it is a non-pre release.
-# If it does not exist, create our own prerelease string
+
 if (EXISTS ${ETH_SOURCE_DIR}/prerelease.txt)
-	file(READ ${ETH_SOURCE_DIR}/prerelease.txt SOL_VERSION_PRERELEASE)
-	string(STRIP "${SOL_VERSION_PRERELEASE}" SOL_VERSION_PRERELEASE)
+	message(FATAL_ERROR "The usage of the file prerelease.txt is deprecated. Please use the file cmake/prerelease.lock instead.")
```

Deprecated sounds as a bit as if it would still be functional for some time. Let's just say it's not supported to make it clear.

And `cmake/prerelease.lock` works in a different way now. We only really kept it for the purpose of creating source archives. I think we should direct users towards the new options instead.

```suggestion
	message(FATAL_ERROR "The usage of the file prerelease.txt is no longer supported. Please use SOL_FORCE_RELEASE option if you want to build a release executable or SOL_PRERELEASE_STRING if you want to adjust the pre-release part of the version string.")
```
**EDIT**: Replaced `SOLC_` prefix with `SOL_` (see comments below).